### PR TITLE
test: control log level with environment variables

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary: We can't change the log level in tests in the past.

### What is changed and how it works?

Add detection for the environment variable `log_level` in each `TestMain()`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -125,24 +125,13 @@ gotest: failpoint-enable
 	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -timeout 20m -cover $(PACKAGES_TIDB_TESTS) -coverprofile=coverage.txt -check.p true > gotest.log || { $(FAILPOINT_DISABLE); cat 'gotest.log'; exit 1; }
 	@$(FAILPOINT_DISABLE)
 
-gotest_in_verify_ci_part_1: failpoint-enable tools/bin/gotestsum tools/bin/gocov tools/bin/gocov-xml
```
Comment (dianqihanwangzi) [inline Makefile:128]: should also remove gocov in br_unit_test_in_verify_ci and dumpling_unit_test_in_verify_ci

File: Makefile [hunk 2]
```diff
@@ -125,24 +125,13 @@ gotest: failpoint-enable
...
-	@echo "Running gotest_in_verify_ci_part_1."
+gotest_in_verify_ci_part_1: failpoint-enable tools/bin/gotestsum
```
Comment (dianqihanwangzi) [inline Makefile:128]: how about change `gotest_in_verify_ci_part_1`  to `gotest_in_verify_ci`?

**Review by dianqihanwangzi — APPROVED** [review]
lgtm

**Review by hawkingrei — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**